libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
I think we are better off with delayed notifications if you consider 80%
use cases, my concern is they get invoked even if chef run fails, hence I
avoid them. 12.5 event handler DSL will impose version requirements, which
is best to avoid, let's roll out the delayed notifications based
implementation and see what feedback we get?
On Nov 11, 2015 10:58 AM, "Nathan Williams" <notifications@github.com>
wrote:

> In libraries/conf.rb
> <https://github.com/nathwill/chef-systemd/pull/69#discussion_r44570656>:
>
> > @@ -98,7 +98,7 @@ def whyrun_supported?
> >          execute "#{r.name}.#{r.conf_type}-systemd-reload" do
> >            command 'systemctl daemon-reload'
> >            action :nothing
> > -          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
> > +          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
> >            subscribes :run, "file[#{conf_path}]", :immediately
>
> ok, so event handler seems really nice. it does still require a recipe be
> included in order to register the callback, but it doesn't rely on
> notifying from every resource...
>
> it's really tempting to remove the daemon-reload from the unit provider,
> and putting the callback registration into the default recipe, but i still
> think 95% of the user-base isn't going to have enough units to worry about
> daemon-reloads, and the advantage of immediately reloading on change seems
> worth the apparent duplication.
>
> regarding requiring chef >= 12.5, i'm kinda torn there too... we already
> require chef >= 12.0, and getting from 11 -> 12 felt like the bigger
> hurdle, so just upping the requirement from 12.0 to 12.5, *if* a user
> wants to take advantage of this use case, doesn't seem like a big thing.
> for whatever subset of users that have to have this feature, and can't
> upgrade chef-client, it's pretty easy to trigger a delayed ruby_block run
> that does the same thing from their own recipe.
>
> does that seem reasonable?
>
> @ranjib <https://github.com/ranjib> also, does this event handler use
> match up with what you were thinking of, or did you have something better
> in mind?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nathwill/chef-systemd/pull/69/files#r44570656>.
>